rust/xaynet-mobile/tests/ffi_test.c

```diff
@@ -74,6 +74,34 @@ static char *test_settings() {
   return 0;
 }
 
+static char *test_global_model() {
+  Settings *settings = xaynet_ffi_settings_new();
+  with_keys(settings);
+  with_url(settings);
+  xaynet_ffi_settings_set_url(settings, "http://localhost:8081");
+
+  Participant *participant = xaynet_ffi_participant_new(settings);
+  mu_assert("failed to create participant", participant != NULL);
+  ModelConfig *model_config = xaynet_ffi_participant_model_config(participant);
+  float* buffer = (float *)malloc(sizeof(float) * model_config->len);
+
+  int err = xaynet_ffi_participant_global_model(NULL, buffer, model_config->data_type, model_config->len);
```

Comment:
Why not have `xaynet_ffi_participant_global_model` take `*ModelConfig` as argument instead of passing the data type and length separately?

Reply:
The `ModelConfig` only describes the model that is expected in the `set_model` function. I didn't want to create the impression that `ModelConfig` belongs to the global model. That was the only reason. I mean the user can use the config as long as the config is never changed. However, it can change if a user restarts the coordinator with a different model len/ mask configuration.
But yeah maybe it is better to use the `ModelConfig`. I will change that

Reply:
if you're worried that `ModelConfig` might cause confusion with that of the global model (and I think that is a valid point), maybe it can be called something else, e.g. `LocalModelConfig`. Then we're sort of saying we expect (or, _hope_) the global model is compatible with our "local model config", although it is not guaranteed as you mentioned in docstrings.

Reply:
Like the idea of `LocalModelConfig` 👍 But if we use this name, I would not change the current signature of the function because it would be confusing if the function uses the local model configuration right?

Reply:
up to you. Passing a `LocalModelConfig` is a bit less brittle, but I understand that this name could be confusing too (but confusing in a _different_ way to if it was just called `ModelConfig`!). It's probably fine if it's documented well. Maybe naming the variable `expected_config` might also help express intent. just a couple of suggestions but happy either way.

Reply:
I changed the name of `ModelConfig` to `LocalModelConfig` but I havn't changed the the function signature. Is it fine if we keep it as it is and change it later?

Reply:
of course, or you can leave as is if that is better for you.